src/metabase/models/permissions.clj

```diff
@@ -367,6 +367,14 @@
 
 ;;; --------------------------------------------------- Helper Fns ---------------------------------------------------
 
+(defn- clear-current-user-cached-permissions!
```

Comment:
Not really part of this fix but I was confused by a test where I wrote something like

```clj
(mt/with-current-user :rasta
  (assert-something-does-not-work)
  (grant-some-permissions!)
  (assert-that-it-now-works))
```

This was because `with-current-user` binds the `*current-user-permissions-set*` delay, and it doesn't get recalculated after the call to `(grant-some-permissions!)`. Rather than rework my test to something like

```clj
(mt/with-current-user :rasta
  (assert-something-does-not-work))
(grant-some-permissions!)
(mt/with-current-user :rasta
  (assert-that-it-now-works))
```

I decided to just go with the better fix and put some code in to reset the bound value of `*current-user-permissions-set*`  so it gets recalculated when permissions change in the current thread.